### ISSUE TITLE
🐛 [Bug] memleak after replace ACE threads

### ISSUE THREAD

**Issue by DSkywalk** [vmangos/core#1098]
## 🐛 Bug report
Since 03a2e108babd7d6049ae1da431e8fe8420d91476 has been accepted, my server consumes all available server memory. This inconsistency causes the server to crash due to lack of memory and other times it simply works but extremely slowly. I have reverted this commit and the server works as expected (see bellow).

![imagen](https://user-images.githubusercontent.com/560310/109791599-e816dc80-7c12-11eb-9c2a-dfbfcdd80bb9.png)

Population: 5 ~ 10 players.

### Expected behavior
My server normally consumes 1GB of RAM.
![imagen](https://user-images.githubusercontent.com/560310/109791677-fcf37000-7c12-11eb-9ba2-def6f6dd23fc.png)

### Steps to reproduce
1. start mangosd 
2. wait and in a few hours the server eats ~3GB

### Version & Environment
Client Version: 1.12.1.5875

Commit Hash:  03a2e108babd7d6049ae1da431e8fe8420d91476

OS Client: Windows
OS Server: debian-jessie and tested on ubuntu 18.04 too.

### Crashlog
- None

Thanks! :+1:

**Comment by ghost** [comment]
@WobLight

**Comment by WobLight** [comment]
@DSkywalk does memory stabilize after a while or does it continue increasing indefinitely? Is the drop in the graph a manual shutdown?

**Comment by DSkywalk** [comment]
does memory stabilize after a while or does it continue increasing indefinitely? 
* It increases until it either crashes or I restart the server.

The image I shared with the problem was a manual shutdown :+1: 

It increases memory usage endlessly even when there are no players on the server and the server is completely empty.

I hope that helps, thanks @WobLight !

**Comment by DSkywalk** [comment]
![imagen](https://user-images.githubusercontent.com/560310/109836851-dfd79500-7c44-11eb-916c-0af5791bee68.png)
I share an general view of the memory consumption since I upgraded to that commit on 24 February. You can also see when the server crashes :bone:

**Comment by WobLight** [comment]
@DSkywalk Are you using stock configuration? Any relevant modifications?

**Comment by DSkywalk** [comment]
my only modification on debian-jessie is to allow compile with cmake 3.0

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 06e77a0..975ca01 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@
 project(MaNGOS)
 
 # CMake policies
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.0)
 
 set(CMAKE_MODULE_PATH
   ${CMAKE_MODULE_PATH}
@@ -34,21 +34,6 @@ option(USE_SCRIPTS "Compile scripts" ON)
 option(USE_EXTRACTORS "Compile extractors" OFF)
 option(USE_LIBCURL "Compile with libcurl for email support" OFF)
 
-find_package(PCHSupport)
-if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.16")
-  option(USE_PCH "Use precompiled headers" ON)
-endif()
-
-if(PCHSupport_FOUND AND (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.16"))
-    if(MSVC)
-      option(USE_PCH_OLD "Use old precompiled headers implementation" ON)
-    else()
-      option(USE_PCH_OLD "Use old precompiled headers implementation" OFF)
-    endif()
-else()
-  unset(USE_PCH_OLD CACHE)
-endif()
-
 if(USE_LIBCURL)
   find_package(CURL REQUIRED)
   #add_definitions(-D_HAS_AUTO_PTR_ETC) # for VS2017 ACE support, won't do any harm to other compilers
@@ -356,8 +341,8 @@ if(UNIX)
     #set(BUILD_ADDITIONAL_FLAGS "${BUILD_ADDITIONAL_FLAGS} -fsanitize=unsigned-integer-overflow")
   endif()
 
-  set(BUILD_ADDITIONAL_FLAGS "${BUILD_ADDITIONAL_FLAGS} --no-warnings -fexceptions -fnon-call-exceptions -march=native -O3 -pipe ")
-  set(CMAKE_CXX_STANDARD 14)
+  set(BUILD_ADDITIONAL_FLAGS "${BUILD_ADDITIONAL_FLAGS} -std=c++14 --no-warnings -fexceptions -fnon-call-exceptions -march=native -O3 -pipe ")
+  #set(CMAKE_CXX_STANDARD 14)
   if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     if(PROFILE_GENERATE)
       set(BUILD_ADDITIONAL_FLAGS "${BUILD_ADDITIONAL_FLAGS} -fprofile-generate")

```

cmake 
```
cmake .. -DCMAKE_INSTALL_PREFIX=/home/vmangos/server -DSERVERS=1 -DTOOLS=0 -DSCRIPTS=1 -DWITH_DOCS=1 -DWITH_COREDEBUG=0

-- Detected 64-bit platform.
-- 
This script builds the MaNGOS server.
  Options that can be used in order to configure the process:
    PREFIX: Path where the server should be installed to
    USE_PCH: Use precompiled headers
    DEBUG: Debug mode
  To set an option simply type -D<OPTION>=<VALUE> after 'cmake <srcs>'.
  For example: cmake .. -DDEBUG=1 -DPREFIX=/opt/mangos

-- Found ACE library: /usr/lib/libACE.so
-- Found ACE headers: /usr/include
-- Using mysql-config: /usr/bin/mysql_config
-- Found MySQL library: /usr/lib/x86_64-linux-gnu/libmysqlclient_r.so
-- Found MySQL headers: /usr/include/mysql
-- 
-- Core revision: d26eafc377e0433d267f 2021-02-19 13:48:27 +0200 
-- Supported version: CLIENT_BUILD_1_12_1
-- Install server to: /home/vmangos/server
-- 
-- Use PCH               : No
-- PGO profile           : Disabled
-- Memory allocation     : TBB/Release /usr/lib/libtbbmalloc.so;/usr/lib/libtbb.so
-- Detected compiler     : GNU
-- Build in debug-mode   : No  (default)
-- Use anticheat    : No  (default)
-- Build scripts    : Yes (default)
-- Debug symbols         : Disabled
-- Build flags (RELEASE) : -O3 -DNDEBUG  -std=c++14 --no-warnings -fexceptions -fnon-call-exceptions -march=native -O3 -pipe 
-- Build flags (DEBUG)   : -g  -std=c++14 --no-warnings -fexceptions -fnon-call-exceptions -march=native -O3 -pipe 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/vmangos/git-tests/build

```

git log

```
commit 763235b0ce93acba86c664425e173a1bfc960245 (HEAD -> skytests)
Author: dskywalk
Date:   Wed Mar 3 06:59:52 2021 +0000

    remove ace, fixed makefile

commit b5111a3ca490f0729fa806977a06075ecb2de04d (origin/development, origin/HEAD, development)
Author: ratkosrb
Date:   Sun Feb 28 19:39:33 2021 +0200

    Fix issues with The Great Masquerade.

```

That's why I tested on ubuntu 18.04 just in case, but it behaves exactly the same.

When I saw the problem I started testing backwards until I came across this commit which was causing the problem. 

thanks!

**Comment by WobLight** [comment]
I meant the mangosd.conf config file, if you send diff over be sure to remove sensible data like database/network config, passwords...

**Comment by DSkywalk** [comment]
my config diffed: https://pastebin.com/udPGVNfS

**Comment by WobLight** [comment]
@DSkywalk please test pr

**Comment by dcolmenero** [comment]
sure, I suggest to use valgrind too :+1: 

```
#!/bin/bash
COUNTER=0
ulimit -c unlimited
while :
do
   valgrind --tool=memcheck --leak-check=full --show-reachable=yes --log-file=valgrind-$COUNTER.log ./mangosd
   mv valgrind.core* valgrind-core.$COUNTER #if system not add .proccess_id to dumps for sure move it to keep it
   COUNTER=$(($COUNTER+1))
done
```

**Comment by WobLight** [comment]
I tried but there was some issue with unrecognized instructions or something. I think it's good now anyway.

**Comment by DSkywalk** [comment]
ok! test started :factory: 

stats of mangosd with ACE threads (commit https://github.com/vmangos/core/commit/03a2e108babd7d6049ae1da431e8fe8420d91476 removed)
![imagen](https://user-images.githubusercontent.com/560310/109954193-55913e80-7ce1-11eb-8157-07f7e9a864be.png)
![imagen](https://user-images.githubusercontent.com/560310/109954205-5924c580-7ce1-11eb-80db-022108aadf43.png)

In a few hours I will share the stats of this PR, thank you very much! :+1:

**Comment by DSkywalk** [comment]
It hasn't been running for many hours yet but from my point of view the problem seems to have been solved with #1099, at this time it should have consumed > 2GB and it is still at ~1GB. 

![imagen](https://user-images.githubusercontent.com/560310/109979342-12de5f00-7cff-11eb-8016-4d255992b1d3.png)
![imagen](https://user-images.githubusercontent.com/560310/109979078-c8f57900-7cfe-11eb-80e6-3563343ff061.png)

So everything seems to be solved now :+1:
